Cabal/Distribution/Types/VersionRange.hs

```diff
@@ -440,17 +441,17 @@ instance Parsec VersionRange where
         verOrWild :: CabalParsing m => m (Bool, Version)
         verOrWild = do
             x <- P.integral
-            verLoop (x :)
+            verLoop (DList.singleton x)
 
         -- trailing: wildcard (.y.*) or normal version (optional tags) (.y.z-tag)
-        verLoop :: CabalParsing m => ([Int] -> [Int]) -> m (Bool, Version)
-        verLoop acc = verLoop' acc <|> (tags >> pure (False, mkVersion (acc [])))
+        verLoop :: CabalParsing m => DList.DList Int -> m (Bool, Version)
+        verLoop acc = verLoop' acc <|> (tags >> pure (False, mkVersion (DList.toList acc)))
```

Comment:
You can also cut down the number of parentheses by rewriting this bit as `mkVersion . DList.toList $ acc`